PowerEditor/src/Parameters.cpp

```diff
@@ -651,38 +651,23 @@ typedef void (WINAPI *PGNSI)(LPSYSTEM_INFO);
 
 void cutString(const TCHAR* str2cut, vector<generic_string>& patternVect)
 {
-	TCHAR str2scan[MAX_PATH];
-	lstrcpy(str2scan, str2cut);
-	size_t len = lstrlen(str2scan);
-	bool isProcessing = false;
-	TCHAR *pBegin = nullptr;
+	const TCHAR *pBegin = str2cut;
+	const TCHAR *pEnd = pBegin;
 
-	for (size_t i = 0 ; i <= len ; ++i)
+	while (*pEnd != '\0')

```

Comment:
What about a nullpointer check for pEnd before dereferencing it?  At least if str2cut is a nullpointer this causes trouble.

Reply:
Fair point - older function would have crashed with null str2cut too but sticking an extra check won't hurt. Should I force push an update or have it as a separate commit? I'm not sure if force push works fine with pull requests.

Reply:
Amend and force push is working for PRs

Reply:
Updated! Thanks.